src/webpack/index.ts

```diff
@@ -0,0 +1,102 @@
+import url from 'url';
+import { Compiler, compilation as webpackCompilation, Module } from 'webpack';
+
+type Compilation = webpackCompilation.Compilation;
+type ModuleExtended = webpackCompilation.Module & {
+  libIdent?: (params: { context: any }) => string;
+  rootModule?: webpackCompilation.Module;
+  rawRequest?: string;
+};
+type Manifest = { [key: string]: { [key: string]: any } };
+type Bundle = {
+  id: number | string;
+  name: string;
+  file: string;
+  publicPath: string;
+};
+
+function buildManifest(compiler: Compiler, compilation: Compilation) {
+  const context = compiler.options.context;
+  const manifest: Manifest = {};
+
+  compilation.chunks.forEach((chunk: webpackCompilation.Chunk) => {
+    chunk.files.forEach(file => {
+      chunk.getModules().forEach((module: ModuleExtended) => {
+        const id = module.id;
+        const name =
+          typeof module.libIdent === 'function'
+            ? module.libIdent({ context })
+            : null;
+        const publicPath = url.resolve(
+          compilation.outputOptions.publicPath || '',
+          file
+        );
+        let currentModule = module;
+
+        if (
+          module.constructor.name === 'ConcatenatedModule' &&
+          module.rootModule
+        ) {
+          currentModule = module.rootModule;
+        }
+
+        if (currentModule.rawRequest && !manifest[currentModule.rawRequest]) {
+          manifest[currentModule.rawRequest] = [];
+        }
+
+        if (
+          currentModule.rawRequest &&
+          Array.isArray(manifest[currentModule.rawRequest])
+        ) {
+          manifest[currentModule.rawRequest].push({
```

Comment:
Does this mean that a given manifest entry will have multiple entries based on the id, name, file, etc? If so, can you provide a test for this so that the behaviour becomes more obvious?

Reply:
This only happens if there are source maps. We don't need them for this purpose however so I will filter them out here instead of inside of the `getBundleFiles` function